Test/c++/LIMITATIONS.md

```diff
@@ -0,0 +1,18 @@
+The C++ backend was written assuming that it would primarily support writing
+C/C++ style code in Dafny, which leads to some limitations in the current
+implementation.
+
+- We do not support BigIntegers, so do not use `int`, or raw instances of
+  `arr.Length`, or sequence length, etc. in executable code.  You can however,
+  use `arr.Length as uint64` if you can prove your array is an appropriate
+  size.  The compiler will report inappropriate integer use.
```

Comment:
Is this a change from a few months ago when (I thought) `int` was compiled as some fixed-width C++ integer type? Or is the new compiler compiling `int` as a fixed-length integer, but this happens to be okay as long as the Dafny program doesn't use `int` (other than temporarily in expressions like `arr.Length as uint64`)?

Does the compiler generate an error if it would have occasion to generate code that would need a `BigInteger`?

Reply:
No, this is how we've consistently used the C++ backend.  We expect the programmer to avoid the use of mathematical integers in their code and instead to use subset types that fit into native types.  That ensures our compilation remains sound.

At the moment, if the C++ backend encounters a BigInteger, it emits a warning and C++ code for a non-existent BigInteger implementation.  We found it useful to warn instead of error, since it gives you a sense of where all of the `int` issues are, rather than just showing you one at a time.

Reply:
Sounds good. That suggests it should be possible to link up the code with a C++ BigInteger library and then run the Dafny-to-C++ compiler on the rest of Dafny's compiler tests.

Reply:
Indeed.  I wasn't sure if Dafny was ready to take a long-term dependency on Boost, GMP (GPL license), or other related library, since C++ doesn't come with its own built-in library, but it's something that could be pursued in the future.  Perhaps if you find an ambitious Amazon intern, they could implement a library in Dafny itself :)

Reply:
I've been insisting to Rustan for a while now that we need to make the compilation process pluggable, so that user codebases can CHOOSE to take such dependencies as they see fit and compile built-in Dafny concepts to whatever implementations they prefer. :) Figuring out the exact plugin interface for that will be tricky, however.